### ISSUE TITLE
Add support for full text search for PostgreSQL

### ISSUE THREAD

**Issue by RobertCraigie** [RobertCraigie/prisma-client-py#52]
[https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search](https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search)

## Suggested solution

Something like the following, `search` should be added to `StringFilter` and must be an instance of `String`.

Should be noted that I'm not stuck on `String` being the name.

```py
# NOTE: actual API is still TODO
from prisma.querying import String

await client.post.find_first(
    where={
        'content': {
            'search': String.contains('cat', 'dog'),
        },
    },
)
await client.post.find_first(
    where={
        'content': {
            # for anything we don't explicitly support
            'search': String.raw('fox \| cat'),
        },
    },
)
```

**Comment by kasem-sm** [comment]
Any updates on this?

**Comment by ntindle** [comment]
subscribing here :)

**Comment by AdeelK93** [comment]
the `String` idea is interesting, but may prove to be challenging to robustly implement, and keep consistent with the TS client.
alternatively, could this be achieved via adding `search` to `StringFilter`?

**Comment by RobertCraigie** [comment]
Yeah `String` could definitely be too involved.

Just supporting a `str` like the TS client does sounds good to me for now. We could always add a more expressive API later if needed.

**Comment by RobertCraigie** [comment]
Implemented in #1003
